src/category_theory/limits/shapes/zero.lean

```diff
@@ -45,6 +45,41 @@ attribute [simp] has_zero_morphisms.comp_zero
 restate_axiom has_zero_morphisms.zero_comp'
 attribute [simp, reassoc] has_zero_morphisms.zero_comp
 
+namespace has_zero_morphisms
+variables {C}
+
+lemma ext' (I J : has_zero_morphisms.{v} C)
```

Comment:
```suggestion
/--
If you're tempted to use this lemma "in the wild", you should probably
carefully consider whether you've made a mistake in allowing two
instances of `has_zero_morphisms` to exist at all.
See, particularly, the note on `zero_morphisms_of_zero_object` below.
-/
lemma ext' (I J : has_zero_morphisms.{v} C)
```

Reply:
I think instead we should just mark this lemma as private. It is completely superseded by the next lemma.